### PR TITLE
[MNG-7600] LocalRepositoryManager is created too early

### PR DESCRIPTION
**Pull request by slawekjaranowski**
LocalRepositoryManager needs configurations from RepositorySystemSession so must be created after session configuration is finished.

---

Following this checklist to help us incorporate your
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue. Your pull request should address just this issue, without
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MNG-XXX] SUMMARY`, where you replace `MNG-XXX`
       and `SUMMARY` with the appropriate JIRA issue. Best practice is to use the JIRA issue
       title in the pull request title and in the first line of the commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.
 - [x] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Review by michael-o — CHANGES_REQUESTED** [review]
to => too

File: maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java [hunk 1]
```diff
@@ -309,6 +288,8 @@ else if ( !MAVEN_RESOLVER_TRANSPORT_AUTO.equals( transport ) )
         mavenRepositorySystem.injectProxy( session, request.getPluginArtifactRepositories() );
         mavenRepositorySystem.injectAuthentication( session, request.getPluginArtifactRepositories() );
 
+        setupLocalRepositoryManager( request, session );
```
Comment (michael-o) [inline maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java:291]: `setUp`. Noun "setup", verb "to set up"

**Review by michael-o — CHANGES_REQUESTED** [review]
The JIRA issue says that properties which are user one are required. MAybe an IT or an example can better desribe the need for?

File: maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java (hunk 1 above)
Comment (slawekjaranowski) [reply maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java]: method renamed

**Comment by slawekjaranowski** [comment]
> The JIRA issue says that properties which are user one are required. MAybe an IT or an example can better desribe the need for?

I know - it will be appreciated to have it tests for such cases ... 

But the problem is that currently it is difficult to mix different Local Manager configurations between parent Maven process which execute tests and Maven process under test. 

Last discussion on the slack ...

**Review by michael-o — APPROVED** [review]
So this basically changes position to have a chance to access all properties...

Does this qualify for 3.9.0?

**Comment by slawekjaranowski** [comment]
> So this basically changes position to have a chance to access all properties...

exactly

> Does this qualify for 3.9.0?

yes, as mentioned in issue

**Pull request merged**